provider/pkg/gen/schema.go

```diff
@@ -614,6 +614,23 @@ type propertyBag struct {
 	requiredSpecs codegen.StringSet
 }
 
+// isDeprecated returns true if the description of a property or a parameter signals that it has
+// been deprecated.
+func isDeprecated(description string) bool {
+	// Unfortunately, there's no fixed format for the deprecation message.
+	// Also, the word "deprecated" appears in several legit descriptions, so
+	// we need that tricky exclusion list below to filter out the actual deprecations.
+	lowerDesc := strings.ToLower(description)
+	return strings.HasPrefix(lowerDesc, "deprecated") ||
+		strings.Contains(lowerDesc, ". deprecated") ||
+		strings.Contains(description, "(Deprecated") ||
+		strings.Contains(description, "Deprecated;") ||
+		strings.Contains(description, "Deprecated.") ||
+		strings.Contains(description, "[Deprecated]") ||
+		strings.Contains(description, "Deprecated in favor") ||
+		strings.Contains(description, "This field is deprecated")
```

Comment:
😭

Reply:
Unfortunately, I feel like we'll have a lot of similar snippets eventually...

Reply:
Though my issue prompted this change, I want to throw some caution here that this is a pretty wide net and will likely catch more than desired. As an example, custom GCE images can be given a `DeprecationStatus`, one of which is - you guessed it - `DEPRECATED`. I would hedge on the side of catching too little rather than too much.

Thanks for the quick work here!

Reply:
Scanning through the file changes, I don't see any that look to be caught in error and I checked the concern above. Nevermind my comment! You guys are awesome 💯

Reply:
> Scanning through the file changes, I don't see any that look to be caught in error and I checked the concern above. Nevermind my comment! You guys are awesome 💯

Thanks for checking. This is definitely the kind of change that should get spidey-senses tingling...

Reply:
I'm happy to tune this rule down to a smaller or a more selective subset of rules that you folks feel would be relatively safe.

Obviously, I'd prefer Google to do this work and have a separate annotation... but there isn't one.

Reply:
Opened https://github.com/pulumi/pulumi-google-native/issues/102 to talk this through with Google folks

Reply:
One other scenario to consider: a new resource is introduced and errantly caught by this filter from the beginning. I don't have a good answer to that aside from:
1. a more concrete indicator upstream of deprecation status
2. generating a reviewable list of all the deprecated and unreviewed attributes.